### PR TITLE
Add a benchmark for the decoder

### PR DESCRIPTION
**Pull request by martinthomson**
This adds a fairly simple varint decoding benchmark to neqo-common.  The next patch will include some improvements to the decoder.

### CODE REVIEW

File: neqo-common/Cargo.toml [hunk 1]
```diff
@@ -28,6 +28,8 @@ qlog = { workspace = true }
 windows = { version = "0.58", default-features = false, features = ["Win32_Media"] }
 
 [dev-dependencies]
+criterion = { version = "0.5", default-features = false }
+neqo-crypto = { path = "../neqo-crypto" }
```
Comment (larseggert) [inline neqo-common/Cargo.toml:32]: I think these should be regular dependencies gated by a `bench` feature, like we do in `neqo-transport/Cargo.toml`. (I'm surprised this compiled...)

Comment (martinthomson) [reply neqo-common/Cargo.toml:32]: It's a dev-dependency, which can't be feature-gated.

File: neqo-common/Cargo.toml [hunk 2]
```diff
@@ -38,3 +40,7 @@ build-fuzzing-corpus = ["hex"]
 [lib]
 # See https://github.com/bheisler/criterion.rs/blob/master/book/src/faq.md#cargo-bench-gives-unrecognized-option-errors-for-valid-command-line-options
 bench = false
+
+[[bench]]
+name = "decoder"
+harness = false
```
Comment (larseggert) [inline neqo-common/Cargo.toml:46]:
```suggestion
harness = false
required-features = ["bench"]
```

Comment (larseggert) [reply neqo-common/Cargo.toml:46]: Also needs `bench = []` under `[features]`.

**Review by larseggert — APPROVED** [review]

**Review by mxinden — APPROVED** [review]

**Pull request merged**